freeipa/src/main/java/com/sequenceiq/freeipa/service/image/userdata/UserDataBuilder.java

```diff
@@ -55,9 +57,26 @@ private String build(Platform cloudPlatform, byte[] cbSshKeyDer, String sshUser,
         model.put("saltBootPassword", saltBootPassword);
         model.put("cbCert", cbCert);
         CcmParameters.addToTemplateModel(ccmParameters, model);
+        extendModelWithProxyParams(proxyConfig, model);
         return build(model);
     }
 
+    private void extendModelWithProxyParams(ProxyConfig proxyConfig, Map<String, Object> model) {
+        if (proxyConfig != null) {
+            model.put("proxyEnabled", Boolean.TRUE);
+            model.put("proxyHost", proxyConfig.getServerHost());
+            model.put("proxyPort", proxyConfig.getServerPort().toString());
+            if (!StringUtils.isEmpty(proxyConfig.getUserName())) {
+                model.put("proxyUser", proxyConfig.getUserName());
+            }
+            if (!StringUtils.isEmpty(proxyConfig.getPassword())) {
+                model.put("proxyPassword", proxyConfig.getPassword());
+            }
```

Comment:
same as above, both or neither of them, maybe it worth adding validation on proxy config creation if doesn't exist